src/main/java/org/opensrp/web/rest/TaskResource.java

```diff
@@ -281,6 +291,25 @@ public void setGson(Gson gson) {
 		modelMap.put("count", countOfTasks != null ? countOfTasks : 0);
 		return new ResponseEntity<>(modelMap, RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
 	}
+
+	@RequestMapping(value = "/search/byCriteria", method = RequestMethod.GET, consumes = {
```

Comment:
Url can not be only marked as `/search` because in this way it is mapping to the existing end-point 
```
@RequestMapping(value = "/{identifier}", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
getByUniqueId method.
``` Thus, resulting in a conflict. Therefore, I have updated the URL to `search/byCriteria.`
Does this sound okay @githengi ?

Reply:
Task Resource does not extend Base resource, this should not be an issue. I have tested /search and its working